File: js/000/TreeNode.js

function TreeNode(val) {
    this.val = val;
    this.left = this.right = null;
}

// int... is -> TreeNode
TreeNode.getTreeNode = function () {
    let tns = new Array(arguments.length);
    for (let i = 0; i < tns.length; i++)
        tns[i] = new TreeNode(arguments[i]);
    for (let i = 0; i < tns.length; i++) {
        let left = 2 * i + 1, right = left + 1;
        if (left >= tns.length) break;
        tns[i].left = tns[left];
        if (right >= tns.length) break;
        tns[i].right = tns[right];
    }
    return tns[0];
}

// TreeNode... tns
TreeNode.printTreeNode = function () {
    for (let tn of arguments)
        console.log(TreeNode.print(tn).join(''));
}

TreeNode.print = function (tn) {
    if (tn == null) return;
    let buf = [];
    buf.push('[');
    Array.prototype.push.apply(buf, TreeNode.print(tn.left));
    buf.push(tn.val);
    Array.prototype.push.apply(buf, TreeNode.print(tn.right));
    buf.push(']');
    return buf;
}

TreeNode.printTreeNode(TreeNode.getTreeNode(1,5,6,2), TreeNode.getTreeNode(7,8,6,2));

File: py/000/TreeNode.py

class TreeNode(object):
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

    @classmethod
    def getTreeNode(cls, *ints):
        tns = [TreeNode(e) for e in ints]
        for i in range(len(tns)):
            left, right = 2*i+1, 2*i+2
            if left >= len(tns):
                break
            tns[i].left = tns[left]
            if right >= len(tns):
                break
            tns[i].right = tns[right]
        return tns[0]

    @classmethod
    def printTreeNode(cls, *nodes):
        for node in nodes:
            print(''.join(TreeNode.p(node)))

    @classmethod
    def p(cls, tn):
        if tn is None:
            return []
        buf = []
        buf.append('[')
        buf += cls.p(tn.left)
        buf.append(str(tn.val))
        buf += cls.p(tn.right)
        buf.append(']')
        return buf

TreeNode.printTreeNode(TreeNode.getTreeNode(1,5,6,2), TreeNode.getTreeNode(7,8,6,2));
